src/opera/commands/deploy.py

```diff
@@ -95,3 +132,18 @@ def deploy(service_template: str, inputs: typing.Optional[dict], storage: Storag
     template = ast.get_template(inputs)
     topology = template.instantiate(storage)
     topology.deploy(num_workers)
+
+
+def prompt_yes_no_question():
+    check = str(input("Do you want to continue? (Y/n): ")).lower().strip()
+    try:
+        if check[:1] == "" or check[:1] == 'y' or check[:1] == "yes":
+            return True
+        elif check[:1] == 'n' or check[:1] == "no":
+            return False
+        else:
+            print('Invalid input. Please try again.')
+            return prompt_yes_no_question()
```

Comment:
Maybe use `str.startswith()` here? Much clearer as to what's really going on at first glance. Also have a list of yes and a list of no options? Something like this:

```python3
def prompt_yes_no_question(yes_responses=("y", "yes"),
                           no_responses=("n", "no"),
                           case_sensitive=False,
                           default_response="yes"):
   ...
```

Reply:
Ok, I've refactored this and used almost similar signature as you suggested. I don't think we need `str.startswith()` because we are now searching for the input in list of yes/no options.